addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,463 @@
+        if (
+            (move.is_sale_document() or move.is_purchase_document())
+            and move.country_code == 'PK'
+            and is_taxable
+        ):
+            return {
+                'post': self._l10n_pk_edi_post_invoice,
+                'cancel': self._l10n_pk_edi_cancel_invoice,
+                'edi_content': self._l10n_pk_edi_xml_invoice_content,
+            }
+
+    def _check_move_configuration(self, move):
+        if self.code != "pk_einvoice_1_3":
+            return super()._check_move_configuration(move)
+        error_messages = []
+        for line in move.invoice_line_ids.filtered(lambda line: line.display_type == 'product'
+            and not self._l10n_pk_is_global_discount(line)):
+            if line.discount < 0:
+                error_messages.append(_("Negative discount is not allowed, set in line %s", line.name))
+            if not line.product_id.l10n_pk_edi_sale_type_id:
+                error_messages.append(_("Sale type is not defined for line %s", line.name))
+            if not line.product_id.l10n_pk_edi_schedule_code_id:
+                error_messages.append(_("Schedule code is not defined for line %s", line.name))
+            hs_code = self._l10n_pk_edi_extract_digits(line.product_id.hs_code)
+            if not hs_code:
+                error_messages.append(_("HS code is not set in product line %s", line.name))
+            elif not re.match("^[a-zA-Z0-9]{8}$", hs_code):
+                error_messages.append(_(
+                    "Invalid HS Code (%(hs_code)s) in product line %(product_line)s",
+                        hs_code=hs_code,
+                        product_line=line.product_id.name or line.name
+                    )
+                )
+        return error_messages
+
+    def _needs_web_services(self):
+        self.ensure_one()
+        return self.code == "pk_einvoice_1_3" or super()._needs_web_services()
+
+    def _l10n_pk_edi_xml_invoice_content(self, invoice):
+        return json.dumps(self._l10n_pk_edi_generate_invoice_json(invoice)).encode()
+
+    def _l10n_pk_validate_seller(self, partner):
+        self.ensure_one()
+        messages = []
+        if not re.match("^.{3,100}$", partner.street or ""):
+            messages.append(_("- Street required min 3 and max 100 characters"))
+        if not partner.street2 and not re.match("^.{3,100}$", partner.street2):
+            messages.append(_("- Street2 should be min 3 and max 100 characters"))
+        if not re.match("^.{3,50}$", partner.state_id.name or ""):
+            messages.append(_("- State required min 3 and max 50 characters"))
+        if messages:
+            action = self.env['res.partner']._get_records_action(name=_("Partner(s)"))
+            error_msg = '\n'.join(map(str, messages))
+            raise RedirectWarning(
+                _("Please enter required details for %s \n%s", partner.display_name, error_msg),
+                action,
+                _('Go to Configuration')
```

Comment:
```suggestion
            error_msg = '\n'.join(map(str, messages))
            raise RedirectWarning(
                _("Please enter required details for %s \n%s", partner.display_name, error_msg),
                action=self.env['res.partner']._get_records_action(name=_("Partner(s)")),
                button_text=_('Go to Configuration')
```

Reply:
Ah, Pylint on runbot tells us it should be 
```py
_("Please enter required details for %(name)s \n%(message)s", name=partner.display_name, message=error_msg),
```

since now for translations we require named args on the translation `_` function.

Reply:
Also on line 128 
![immagine](https://github.com/user-attachments/assets/cbfcf9fd-db9f-4c88-aede-3b6083e007cd)